corehq/sql_db/management/commands/locate_invalid_shard_data.py

```diff
@@ -1,20 +1,37 @@
-from __future__ import absolute_import, print_function
-from __future__ import unicode_literals
+from __future__ import absolute_import, print_function, unicode_literals
+
 from django.core.management.base import BaseCommand
-from corehq.sql_db.shard_data_management import get_count_of_unmatched_models_by_shard
-from corehq.sql_db.util import get_db_aliases_for_partitioned_query, get_all_sharded_models
+
+from corehq.sql_db.shard_data_management import (
+    delete_unmatched_shard_data,
+    get_count_of_unmatched_models_by_shard,
+)
+from corehq.sql_db.util import (
+    get_all_sharded_models,
+    get_db_aliases_for_partitioned_query,
+)
 
 
 class Command(BaseCommand):
     help = "Print out all shard data that exists in databases that don't contain the associated shards."
 
+    def add_arguments(self, parser):
+        parser.add_argument('--delete', action='store_true')
```

Comment:
perhaps should add a database argument as well

Reply:
sounds useful